src/cpu/aarch64/jit_aarch64_sve_512_conv_kernel.hpp

```diff
@@ -0,0 +1,685 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_JIT_SVE_CONV_KERNEL_HPP
+#define CPU_AARCH64_JIT_SVE_CONV_KERNEL_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/memory_tracking.hpp"
+
+#include "cpu/aarch64/jit_generator.hpp"
+#include "cpu/aarch64/jit_primitive_conf.hpp"
+
+#define DISABLE_ELTWISE
+#ifndef DISABLE_ELTWISE
+#include "cpu/aarch64/jit_uni_eltwise_injector.hpp"
+#endif // #ifndef DISABLE_ELTWISE
+
+#define PRFWMAX 31
+#define LDRMAX 255
+#define LDRWMAX 252
+#define ADDMAX 4095
+#define PRFMMAX 32760
+#define MOVMAX 65535
+/* Get vector offsets, ofs / VL(VL: 512bits = 64Bytes) */
+#define VL_OFS(ofs) ((ofs) >> 6)
+
+using namespace Xbyak_aarch64;
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+template <typename Vmm>
+struct _jit_aarch64_sve_512_conv_fwd_kernel : public jit_generator {
+
+    _jit_aarch64_sve_512_conv_fwd_kernel(
+            const jit_conv_conf_t &ajcp, const primitive_attr_t &attr)
+#ifndef DISABLE_ELTWISE
+        : jcp(ajcp), attr_(attr), eltwise_injector_(nullptr) {
+#else // #ifndef DISABLE_ELTWISE
+        : jcp(ajcp), attr_(attr) {
+#endif // #ifndef DISABLE_ELTWISE
+
+        if (jcp.with_eltwise)
+#ifndef DISABLE_ELTWISE
+            eltwise_injector_ = new jit_uni_eltwise_injector_f32<avx512_common>(
```

Comment:
Just wondering if we need to substitute `avx512_common ---> sve_512` though as I understand currently `DISABLE_ELTWISE` is set so it does not really matter. Nevertheless it looks bit strange to see `x64` isa here!

Reply:
Thank you for pointing it out. It just forgot to fix it.

Reply:
Looks good now, thanks!